### PR TITLE
Compute near duplicates + ROI fields

### PR DESCRIPTION
**Pull request by brimoor**
## Change log

- Adds a `fob.compute_near_duplicates()` method that provides a use case-centric interface to the existing `DuplicatesMixin.find_duplicates()` method
- Also adds `roi_field` arguments to `compute_similarity()` and `compute_leaky_splits()` for consistency with `compute_uniqueness()` and `compute_representativeness()`

## Example near duplicates usage

```py
import fiftyone as fo
import fiftyone.brain as fob
import fiftyone.zoo as foz

# Load some COCO data
dataset = foz.load_zoo_dataset(
    "coco-2017",
    split="validation",
    max_samples=1000,
    persistent=True,
)

# Compute embeddings
model = foz.load_zoo_model("mobilenet-v2-imagenet-torch")
dataset.compute_embeddings(model, embeddings_field="embeddings")

# Find near duplicates
index = fob.compute_near_duplicates(dataset, embeddings="embeddings")

print(index.thresh)  # 0.2
print(index.duplicate_ids)  # ['XXXX', ...]

duplicates = index.duplicates_view()

session = fo.launch_app(duplicates)
```

### Setup

```py
import fiftyone as fo
import fiftyone.brain as fob
import fiftyone.zoo as foz
import fiftyone.utils.random as four

dataset = foz.load_zoo_dataset("quickstart")

# model = foz.load_zoo_model("mobilenet-v2-imagenet-torch")
# dataset.compute_patch_embeddings(model, "ground_truth", embeddings_field="embeddings")
```

###  Similarity w/ ROI

```py
index = fob.compute_similarity(
    dataset[:100],
    roi_field="ground_truth",
    # embeddings="embeddings",
)

print(index.total_index_size)  # 100

embeddings, sample_ids, _ = index.compute_embeddings(dataset[100:])
index.add_to_index(embeddings, sample_ids)

print(index.total_index_size)  # 200
```

### Near duplicates w/ ROI

```py
index = fob.compute_near_duplicates(
    dataset,
    roi_field="ground_truth",
    # embeddings="embeddings",
)

print(index.thresh)  # 0.2
print(index.duplicate_ids)  # ['XXXX', ...]

duplicates = index.duplicates_view()
```

### Leaky splits w/ ROI

```py
dataset.untag_samples(dataset.distinct("tags"))
four.random_split(dataset, {"train": 0.7, "test": 0.3})
index = fob.compute_leaky_splits(
    dataset,
    ["train", "test"],
    roi_field="ground_truth",
    # embeddings="embeddings",
)

print(index.thresh)  # 0.2
print(index.leak_ids)  # ['XXXX', ...]

leaks = index.leaks_view()
```

### CODE REVIEW

File: fiftyone/brain/internal/core/duplicates.py [hunk 1]
```diff
@@ -10,13 +10,87 @@
...
+            skip_failures=skip_failures,
+            progress=progress,
+        )
+    elif not isinstance(similarity_index, fbs.DuplicatesMixin):
+        raise ValueError(
+            "This method only supports similarity indexes that implement the "
+            "%s mixin" % fbs.DuplicatesMixin
+        )
+
+    similarity_index.find_duplicates(thresh=threshold)
```
Comment (jacobsela) [inline fiftyone/brain/internal/core/duplicates.py:90]: @brimoor This is what I mean with embeddings/models/similarity boilerplate. This function is 60 lines of code for effectively one line.

File: fiftyone/brain/internal/core/leaky_splits.py [hunk 2]
```diff
@@ -30,18 +30,21 @@ def compute_leaky_splits(
     samples,
     splits,
     threshold=None,
+    roi_field=None,
```
Comment (jacobsela) [inline fiftyone/brain/internal/core/leaky_splits.py:33]: I like this for copy-paste augments and other strange edge cases.

File: fiftyone/brain/similarity.py [hunk 3]
```diff
@@ -32,13 +32,21 @@
...
+)
+
 _DEFAULT_MODEL = "mobilenet-v2-imagenet-torch"
 _DEFAULT_BATCH_SIZE = None
 
 
 def compute_similarity(
     samples,
     patches_field,
+    roi_field,
```
Comment (jacobsela) [inline fiftyone/brain/similarity.py:49]: what's the difference between ROI field and patches field from a functional standpoint?

File: fiftyone/brain/internal/core/duplicates.py (hunk 1 above)
Comment (brimoor) [reply fiftyone/brain/internal/core/duplicates.py:90]: Yeah but this boilerplate does serve some purposes:
- Allow `compute_near_duplicates()` to use a different default model than `compute_similarity()`
- Enforce that a pre-existing `similarity_index` must implement the `DuplicatesMixin`

File: fiftyone/brain/similarity.py (hunk 3 above)
Comment (jacobsela) [reply fiftyone/brain/similarity.py:49]: So in that case passing a single patch per sample to patches_field is equivalent to ROI field? feels strange to have them as two different arguments.

**Review by jacobsela — APPROVED** [review]

**Pull request merged**